### PR TITLE
Update Go SDK to v1.26.0-rc.2

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
## What changed?
Update Go SDK to v1.26.0-rc.2

## Why?
Use a tagged Go SDK for the server release

### CODE REVIEW

**Review by bergundy — COMMENTED** [review]
Can we hold off on this until we merge the nexus feature branch?
All of the fixes you made are addressed there and will generate conflicts.

**Comment by Quinn-With-Two-Ns** [comment]
>Can we hold off on this until we merge the nexus feature branch?

Doesn't matter to me. I was under the impression we wanted to cherry pick these changes to the latest release candidate, merging on top of the nexus feature branch would make that hard.

**Review by Ahmed-aleryani — APPROVED** [review]

**Comment by tdeebswihart** [comment]
> Can we hold off on this until we merge the nexus feature branch? All of the fixes you made are addressed there and will generate conflicts.

This is now unfortunately blocking my work on our OpenAPI specs; do you have a timeline for this? The OpenAPI work isn't urgent

**Comment by vinimona** [comment]
I have another [PR](https://github.com/temporalio/temporal/pull/5406) to update the otelgrpc go package. I didn't realize that this PR covers that change. Do you have an estimate on when this will be available?  Thanks.

**Comment by Quinn-With-Two-Ns** [comment]
It was decided we won't pick up `api 1.27.0` or the go sdk `v1.26.0-rc.2` for the Server release I am going to close this PR since it isn't needed at the moment and will conflict with nexus and open up a PR to update the Server once the go SDK  `v1.26.0` if released and the nexus work is merged.

**Comment by bergundy** [comment]
Reopened since we plan on delaying the nexus feature branch merge.

File: service/frontend/configs/quotas.go
Comment (tdeebswihart) [inline service/frontend/configs/quotas.go:144]: @bergundy is this correct?

**Review by tdeebswihart — APPROVED** [review]

File: service/frontend/configs/quotas.go
Comment (bergundy) [reply service/frontend/configs/quotas.go:144]: Don't worry about this, these APIs are not implemented.

**Pull request merged**
